### ISSUE TITLE
Having Comments as Child Nodes in container breaks DOM Driver

### ISSUE THREAD

**Issue by FeliciousX** [cyclejs/cyclejs#561]
If the HTML container contains comments as children element DOM Driver would error.

It works if it matches exactly the container.

**Code to reproduce the issue:**
https://esnextb.in/?gist=026302a35432e86020e13104b034d0f0

```html
<div id="app">
  <!-- boom! -->
</div>
```

```javascript
function main() {
  const sinks = {
    DOM: xs.periodic(1000).map(i =>
      // this breaks
      h('h1', '' + i + ' seconds elapsed')
  
      // this works
      // h('div', { props: { id: 'app' }}, [ h('h1', '' + i + ' seconds elapsed') ])
    )
  };
  return sinks;
}
```

**Expected behavior:**
It replaces the container and just runs normally

**Actual behavior:**
Errors out :(

**Versions of packages used:**

```json
{
  "dependencies": {
    "xstream": "10.2.0",
    "@cycle/run": "3.0.0",
    "@cycle/dom": "16.0.0"
  }
}
```

**Comment by staltz** [comment]
```json
{
  "dependencies": {
    "xstream": "10.3.0",
    "@cycle/run": "3.0.0",
    "@cycle/dom": "16.0.0"
  }
}
```

https://www.webpackbin.com/bins/-KfkKF7Q9Oq58Uou9yeD

**Comment by FeliciousX** [comment]
sorry, updated the `gist`, `title` and `package`

**Comment by FeliciousX** [comment]
only happens if it's `comments` but not if it's a normal element

**Comment by staltz** [comment]
Okay, thanks for the hint. I'll mark it `should` since a comment isn't a showstopper.

**Comment by staltz** [comment]
I sent a PR to snabbdom, pending on that for this issue to be fixed.

**Comment by FeliciousX** [comment]
thanks @staltz  :D

**Comment by staltz** [comment]
This was merged and solved in snabbdom v0.6.8, and we'll soon release a Cycle DOM version with it.
